synthetics_mirrors.tf

```diff
@@ -23,11 +23,42 @@ resource "datadog_synthetics_test" "mirrors_jenkins_io" {
   status = "live"
 }
 
+resource "datadog_synthetics_test" "mirrors_jenkins_io_enforced_https" {
+  type = "api"
+  request_definition {
+    method = "GET"
+    url    = "http://mirrors.jenkins.io"
+  }
+  assertion {
+    type     = "statusCode"
+    operator = "is"
+    target   = "308"
+  }
+  assertion {
+    type     = "header"
+    property = "location"
+    operator = "is"
+    target   = "https://mirrors.jenkins.io"
+  }
+  locations = ["aws:eu-central-1"]
+  options_list {
+    tick_every = 900
+  }
+  name    = "mirrors.jenkins.io-enforced-http"
+  message = "Notify @pagerduty"
+  tags = [
+    "jenkins.io",
+    "production"
+  ]
+
+  status = "live"
+}
+
 resource "datadog_synthetics_test" "mirrors_jenkinsci_org" {
   type = "api"
   request_definition {
     method = "GET"
-    url    = "https://mirrors.jenkins-ci.org"
+    url    = "https://mirrors.jenkins-ci.org/debian/jenkins_2.251_all.deb?mirrorlist"
```

Comment:
Will this version have to be updated in the future?

Reply:
nope: a given version must always be available through the mirrors.
The idea is to monitor the whole service (e.g. "does it work as expected?") and not only the HTTP answer. By providing a full URL to an existing file, with the query string parameter `mirrorlist`, then we expect the HTTP/200 page.
Since mirrorbits does not have the file itself (it is expected to answer an HTTP/3XX redirect to a mirror), an HTTP/200 only means that we have the full-featured service.

=> this was the behavior of the monitor on get.jenkins.io, so this change is to make it coherent (as the 3 domains `get.jenkins.io`, `mirrors.jenkins.io` and `mirrors.jenkins-ci.org` are pointing towards the same service)